### ISSUE TITLE
App closes after boarding completed since latest GSX Update

### ISSUE THREAD

**Issue by compumark** [Fragtality/Fenix2GSX#86]
Hi,
the app closes it self after the boarding is completed. The log says 
2024-08-06 08:12:17.405 [INF] [ GsxController:RunDepartureServic ] Transmitting Final Loadsheet ... 
2024-08-06 08:12:17.405 [DBG] [ GsxController:RunDepartureServic ] SeatMap updates inhibited (finalLoadsheetTriggered) 

But a loadsheet was never delivered. I checked the app and found that its not running anymore. 
In the Windows event viewer the following occurs:

![image](https://github.com/user-attachments/assets/e1b93ccf-e3bc-4ab1-96de-c3dbf77603eb)

![image](https://github.com/user-attachments/assets/e9b54472-e4cf-4ac4-bdc5-7ef8f28e4352)

![image](https://github.com/user-attachments/assets/95ee1869-df43-4cb6-97e5-df16b8314944)

[Fenix2GSX20240806.log](https://github.com/user-attachments/files/16506133/Fenix2GSX20240806.log)

The only thing that changed was the GSX update from 3.0.8 to 3.0.9. With the flight yesterday, where it was working GSX 3.0.8 was installed.

It was initially observed yesterday and replicated today.
Thanks a lot for any help,
Markus

**Comment by compumark** [comment]
Yes, I doubt it as well, as I have cross checked with other people, they are using GSX 2.0.9 and Fenix2GSX 0.4.10 and dont reseive the error.

I also resetted the Aircraft profile to default before trying again, didnt worked for me. 

I will try this after landing, just in cruise :-) - I will report back

**Comment by compumark** [comment]
I installed the update now and restarted the computer. Sadly its not working.
I tried it two times. 

![image](https://github.com/user-attachments/assets/2e9bd2b5-798e-4a28-9e4e-3c9787ce17d6)

I then stopped the time from the Final loadsheet time posted in the app until the crash and its around that second amount.

![image](https://github.com/user-attachments/assets/e70ebc17-0eb4-452b-9a1e-02a358379560)
![image](https://github.com/user-attachments/assets/62294b31-94f2-4b16-96c3-40ee010bfc20)

This are the last lines in the log file:
2024-08-06 17:11:29.667 [DBG] [ FenixSeatMap:BoardPassengers     ] (delta 0 | gsxTotal 53 | current 53 | actual 54 | planned (50) 
2024-08-06 17:11:30.828 [DBG] [ FenixSeatMap:BoardPassengers     ] (delta 0 | gsxTotal 53 | current 53 | actual 54 | planned (50) 
2024-08-06 17:11:31.975 [DBG] [ FenixSeatMap:BoardPassengers     ] (delta 1 | gsxTotal 54 | current 53 | actual 54 | planned (50) 
2024-08-06 17:11:31.976 [INF] [ GsxController:RunLoadingServices ] Aircraft loaded 
2024-08-06 17:11:34.236 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Aft Door (False) 
2024-08-06 17:11:34.487 [INF] [ GsxController:RunServices        ] Boarding completed 
2024-08-06 17:11:34.488 [INF] [ GsxController:RunDepartureServic ] Final Loadsheet in 149s 
2024-08-06 17:11:37.920 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Aft Door (False) 
2024-08-06 17:11:42.730 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Aft Door (False) 
2024-08-06 17:11:53.253 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Fwd Door (False) 
2024-08-06 17:11:56.951 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Fwd Door (False) 
2024-08-06 17:12:01.743 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Fwd Door (False) 
2024-08-06 17:13:01.480 [INF] [ GsxController:RunDepartureServic ] Transmitting Final Loadsheet ... 
2024-08-06 17:13:01.480 [DBG] [ GsxController:RunDepartureServic ] SeatMap updates inhibited (finalLoadsheetTriggered)

**Comment by Fragtality** [comment]
Seems like something on your System is interfering (AV / Security-Software? Something that blocks Chrome?) 🤔 

Please share the `debug.log` from the bin-Directory.

**Comment by compumark** [comment]
No, not really. It was all working until yesterday. In this time there were no windows updates deployed.

[debug.log](https://github.com/user-attachments/files/16512377/debug.log)

**Comment by Fragtality** [comment]
Does that make any Difference?
https://www.mediafire.com/file/1xfwyzy1hyu6qk0/Fenix2GSX-Installer-v0.4.11-dev.exe/file

**Comment by Fragtality** [comment]
Ok, that Version is at least starting ^^
https://www.mediafire.com/file/3uvpa49efrtheji/Fenix2GSX-Installer-v0.4.11-dev.exe/file

(I'm just fishing in the Dark here ... basically your Copy crashes when Chromium is loaded in a Way which does not even produce an Exception to catch)

**Comment by compumark** [comment]
No luck, but one line more in the log file:

2024-08-06 20:26:22.139 [INF] [ GsxController:RunDepartureServic ] Transmitting Final Loadsheet ... 
2024-08-06 20:26:22.139 [DBG] [ GsxController:RunDepartureServic ] SeatMap updates inhibited (finalLoadsheetTriggered) 
2024-08-06 20:26:22.139 [DBG] [ FenixEFBInterface:TriggerFinalOn ] Connecting to EFB ... 

the debug.log shows
[0806/201617.597:WARNING:resource_util.cc(99)] Please customize CefSettings.root_cache_path for your application. Use of the default value may lead to unintended process singleton behavior.

I then started the Fenix2GSX as Admin as a trial, but same behavoir. I checked the Windows Security and can confirm that the whole directory is excluded.
![image](https://github.com/user-attachments/assets/0e28cdbb-0460-4c83-a458-01135b44fb84)

**Comment by Fragtality** [comment]
> No luck, but one line more in the log file:

Yeah, but it confirms where the Crash is happening - it silently & completely crashes when Chromium should open the EFB.

<br/>

> the debug.log shows

Well that Message is always there and it was never an Issue before (also for your Setup). I could try to set this Path in another Build, just to have it excluded as Cause.

<br/>

> I then started the Fenix2GSX as Admin as a trial, but same behavoir. I checked the Windows Security and can confirm that the whole directory is excluded.

At least that answers the Question if running as Admin was the Cause of the Issue - no it isn't and it is also not the Solution 😅 

I don't want to make any false Hope: I just don't see how I can fix that. There is nothing in the Code to fix, else I would have the Exception Stack by now. Something "below" in the Framework/Libraries/System does not want to run the Browser Engine and that is beyond my Scope.
I could look into if I can add an Option to disable the Code which is using CEF. Which would considerably alter the Behavior, but at least you can continue to use Fenix2GSX - because the only alternative currently would be to not use it anymore :/

**Comment by Fragtality** [comment]
So here is the new Build: https://www.mediafire.com/file/ntxd74wrlrdi62j/Fenix2GSX-Installer-v0.4.11.exe/file

I've set the "root_cache_path" Cef was warning about - which leads to much clutter in the bin Directory (so it might make that an Option too).
Try if that helps in your Situation. Maybe it logs more about the Error if it still happens, as there is now a chrome_debug.log File in the bin Directory.

If that first Test was not successful, please set "skipCef" to true in the %appdata%\Fenix2GSX\Fenix2GSX.config File. This disables Calls where CEF is used. Be aware that you will not receive a Final LS (although Fenix2GSX says so) - it is just a quick-fix.

**Comment by paulwain** [comment]
Just letting you know I am also having the above issues ^^ since the latest GSX update :( Apps keep crashing when boarding is complete. Worked fine before the update.

**Comment by Fragtality** [comment]
@compumark @paulwain
https://www.mediafire.com/file/5ybja7cek6ihewf/Fenix2GSX-Installer-v0.4.11.exe/file
Sorry, must have mixed up the Links!

> Just letting you know I am also having the above issues ^^ since the latest GSX update :( Apps keep crashing when boarding is complete. Worked fine before the update.

There won't be a Solution - only this ugly Quick-Fix.
It is working fine for others after the Update, it is not a Crash due to bugged Code in the App, Fenix only maybe has a Solution in the Future (not forcing me to have to talk to the EFB) and Umberto certainly won't into that.

**Comment by paulwain** [comment]
I have downloaded and will test it out later :) So the only thing I wont get is the loadsheet with this fix ^ Do i need to make any changes or can I just update ? 

Thank you

**Comment by Fragtality** [comment]
@paulwain @compumark 
1 - Just run the new Version and test if it behaves differently (only relevant for Paul - for Mark we already established that the Issue was not outdated Library Versions)
2 - Run the Tool with the Setting 'setCefRootPath' set to 'true' in the Config File. (If not successful, share the chrome_debug.log File from the bin Directory)
3 - Run the Tool with the Setting 'skipCef' set to 'true' in the Config File. (setCefRootPath can be set back to false then)

These Settings are only available in the  %appdata%\Fenix2GSX\Fenix2GSX.config File. (Tool needs to be restarted when Config File was edited)

Point 3 would be the mentioned ugly Quick-Fix to prevent these silent Crashes. It is up to you both if that is something you two can work with, i.e. if the Tool is still useable for you.
The Problem with Fenix V2 EFB is that the Loading State and the Loadsheets are tightly coupled - so it is not only the Prelim/Final LS missing as ACARS Message, it is also that the EFB can not offer to copy the Final Values for TO Calculation and the EFB doesn't consider the Loading as finished. I can't name the exact Consequences or Issues that rise through this Quick-Fix - the Tool just isn't meant to be run this Way. It is the best I can do to prevent it at least from Crashing :/

When it comes to EFB internals and GSX, V2 wasn't an Improvement (more of the Contrary tbh ...)

**Comment by compumark** [comment]
I just installed the version above and started it with no config change, same issue occurs. After landing I will try step 2

**Comment by compumark** [comment]
I have now added the following line:   <add key="setCefRootPath" value="true" /> in the Fenix2GSX.config file and restarted the application and it was working now. No crash anymore.
[chrome_debug.log](https://github.com/user-attachments/files/16558728/chrome_debug.log)

and the log from the application

2024-08-09 10:51:24.443 [INF] [ GsxController:RunLoadingServices ] Aircraft loaded 
2024-08-09 10:51:24.443 [DBG] [ FenixController:DoorAft          ] Triggering Left Aft Door (False) 
2024-08-09 10:51:24.703 [DBG] [ GsxController:MenuOpen           ] Opening GSX Menu 
2024-08-09 10:51:24.704 [INF] [ GsxController:RunLoadingServices ] Removing Aft Stairs 
2024-08-09 10:51:25.030 [DBG] [ GsxController:MenuWaitReady      ] Wait ended after 300ms 
2024-08-09 10:51:25.030 [DBG] [ GsxController:MenuItem           ] Selecting Menu Option 7 (L-Var Value 6) 
2024-08-09 10:51:26.531 [INF] [ GsxController:RunDepartureServic ] Final Loadsheet in 101s 
2024-08-09 10:51:28.974 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Aft Door (False) 
2024-08-09 10:51:32.932 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Aft Door (False) 
2024-08-09 10:51:45.506 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Fwd Door (False) 
2024-08-09 10:51:49.485 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Fwd Door (False) 
2024-08-09 10:51:53.419 [DBG] [ FenixController:DoorFwd          ] Triggering Cargo Fwd Door (False) 
2024-08-09 10:52:30.547 [INF] [ GsxController:RunDepartureServic ] Transmitting Final Loadsheet ... 
2024-08-09 10:52:30.548 [DBG] [ GsxController:RunDepartureServic ] SeatMap updates inhibited (finalLoadsheetTriggered) 
2024-08-09 10:52:30.548 [DBG] [ FenixEFBInterface:TriggerFinalOn ] Connecting to EFB ... 
2024-08-09 10:52:31.319 [DBG] [ FenixEFBInterface:EfbHandlerFina ] Setting EFB ... 
2024-08-09 10:52:55.666 [DBG] [ GsxController:DingCabin          ] Ding-ing the Cabin 
2024-08-09 10:52:55.974 [INF] [ GsxController:RunDepartureServic ] Final Loadsheet sent via ACARS 
2024-08-09 10:52:55.975 [DBG] [ GsxController:CloseDoorsOnFinal  ] Closing Fwd/Aft Doors 
2024-08-09 10:52:55.975 [DBG] [ FenixController:DoorFwd          ] Triggering Left Fwd Door (False) 
2024-08-09 10:52:57.398 [DBG] [ GsxController:RunServices        ] SeatMap updates allowed (!finalLoadsheetTriggered) 
2024-08-09 10:52:58.621 [DBG] [ FenixSeatMap:Update              ] Planned Passenger changed to 22! 
2024-08-09 10:52:58.621 [DBG] [ FenixSeatMap:InitSeatMaps        ] count: 22 | len: 180 
2024-08-09 10:52:58.621 [DBG] [ FenixSeatMap:InitSeatMaps        ] Seat #103 is an Extra! 
2024-08-09 10:52:58.621 [DBG] [ FenixSeatMap:InitSeatMaps        ] Seat #127 is an Extra! 
2024-08-09 10:52:58.621 [DBG] [ FenixSeatMap:InitSeatMaps        ] Seat #152 is an Extra! 
2024-08-09 10:52:58.621 [DBG] [ FenixSeatMap:InitSeatMaps        ] Difference to planned Pax: 3 
2024-08-09 10:52:58.621 [DBG] [ FenixSeatMap:InitSeatMaps        ] Difference to planned Cargo: 45kg / 99,225lbs 
2024-08-09 10:53:06.978 [INF] [ GsxController:RunDepartureServic ] Preparing for Pushback - removing Equipment (Beacon is on) 
2024-08-09 10:53:06.978 [DBG] [ GsxController:MenuOpen           ] Opening GSX Menu 
2024-08-09 10:53:06.979 [INF] [ GsxController:RunDepartureServic ] Removing Jetway 
2024-08-09 10:53:07.305 [DBG] [ GsxController:MenuWaitReady      ] Wait ended after 300ms 
2024-08-09 10:53:07.305 [DBG] [ GsxController:MenuItem           ] Selecting Menu Option 6 (L-Var Value 5) 
2024-08-09 10:53:07.568 [INF] [ GsxController:RunDepartureServic ] Disconnecting GPU (forced False | manual: False) 
2024-08-09 10:53:07.582 [DBG] [ MobiSimConnect:SimConnect_OnRece ] TOGGLE_JETWAY Event received (dwId 4 | dwData 0 | IsJetwayConnected True) 
2024-08-09 10:53:11.231 [DBG] [ GsxController:CheckIntRadSwitch  ] INT/RAD Switch was flicked to INT Position 
2024-08-09 10:53:11.231 [INF] [ GsxController:RunDepartureServic ] Calling Push-Back Service (INT/RAD Switch) 
2024-08-09 10:53:11.231 [DBG] [ GsxController:MenuOpen           ] Opening GSX Menu 
2024-08-09 10:53:11.667 [DBG] [ GsxController:MenuWaitReady      ] Wait ended after 400ms 
2024-08-09 10:53:11.667 [DBG] [ GsxController:MenuItem           ] Selecting Menu Option 5 (L-Var Value 4) 
2024-08-09 10:53:11.668 [DBG] [ GsxController:ResetIntRadSwitch  ] Resetting INT/RAD Switch to OFF Position 
2024-08-09 10:53:12.879 [INF] [ GsxController:RunDepartureServic ] Push-Back Service is active

**Comment by paulwain** [comment]
> @paulwain @compumark 1 - Just run the new Version and test if it behaves differently (only relevant for Paul - for Mark we already established that the Issue was not outdated Library Versions) 2 - Run the Tool with the Setting 'setCefRootPath' set to 'true' in the Config File. (If not successful, share the chrome_debug.log File from the bin Directory) 3 - Run the Tool with the Setting 'skipCef' set to 'true' in the Config File. (setCefRootPath can be set back to false then)
> 
> These Settings are only available in the %appdata%\Fenix2GSX\Fenix2GSX.config File. (Tool needs to be restarted when Config File was edited)
> 
> Point 3 would be the mentioned ugly Quick-Fix to prevent these silent Crashes. It is up to you both if that is something you two can work with, i.e. if the Tool is still useable for you. The Problem with Fenix V2 EFB is that the Loading State and the Loadsheets are tightly coupled - so it is not only the Prelim/Final LS missing as ACARS Message, it is also that the EFB can not offer to copy the Final Values for TO Calculation and the EFB doesn't consider the Loading as finished. I can't name the exact Consequences or Issues that rise through this Quick-Fix - the Tool just isn't meant to be run this Way. It is the best I can do to prevent it at least from Crashing :/
> 
> When it comes to EFB internals and GSX, V2 wasn't an Improvement (more of the Contrary tbh ...)

@Fragtality - I can also confirm point #2 works - Just did 2 flights and works. Thank you for the quick fix :)

**Comment by Fragtality** [comment]
Wow, ok, great - that is a bit of a surprise 😅 
I still don't understand it really, but when it solves the Problem without any drawback then it is even better!

**Comment by Fragtality** [comment]
Closed => Fixed in 0.4.11 Release when setting 'setCefRootPath' set to 'true'
